### PR TITLE
feat: adds ability to provide redirect uri

### PR DESCRIPTION
**Pull request by chalmerlowe**
WIP PR for discussion: aiming to provide the ability to include a redirect URI, client ID, and client secrets to facilitate the migration away from "out of band" OAuth authentication.

@tswast 

For context, there are changes in these repos:
* https://github.com/googleapis/python-bigquery-pandas/pull/595 #python-bigquery-pandas
* https://github.com/pydata/pydata-google-auth/pull/58 (Non-google repository)

### CODE REVIEW

**Comment by chalmerlowe** [comment]
These failing tests are due to the fact that pydata-google-auth also needs a matching update and that PR (noted above) has not been merged into the main project yet. When that is merged, the expectation is that these tests will pass.

File: docs/_static/js/authcodescripts.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+function onloadoauthcode() {
```
Comment (tswast) [inline docs/_static/js/authcodescripts.js:1]: Nit: We'll need a license header here. pandas-gbq inherited a BSD license, so it'll look a bit different from the Apache 2.0 header.

```suggestion


// Copyright (c) 2023 pandas-gbq Authors All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.


function onloadoauthcode() {
```

File: pandas_gbq/auth.py [hunk 2]
```diff
@@ -43,10 +49,11 @@ def get_credentials(
 
     credentials, default_project_id = pydata_google_auth.default(
         SCOPES,
-        client_id=CLIENT_ID,
-        client_secret=CLIENT_SECRET,
+        client_id=client_id,
```
Comment (tswast) [inline pandas_gbq/auth.py:52]: We need to keep our default client ID and secret.

Note: if `auth_local_webserver = False`, then ideally we provide a different default client ID & secret & a default redirect URI, but we can/should do that in a follow-up PR.

File: pandas_gbq/gbq.py [hunk 3]
```diff
@@ -864,6 +877,8 @@ def read_gbq(
         or
         :func:`google.oauth2.service_account.Credentials.from_service_account_file`
         instead.
+    # TODO: add parameters for auth_redirect_uri, client_id, client_secret
```
Comment (tswast) [inline pandas_gbq/gbq.py:880]: Need to complete this TODO

File: pandas_gbq/gbq.py [hunk 4]
```diff
@@ -912,6 +927,9 @@ def read_gbq(
         credentials=credentials,
         private_key=private_key,
         use_bqstorage_api=use_bqstorage_api,
+        auth_redirect_uri=auth_redirect_uri,
```
Comment (tswast) [inline pandas_gbq/gbq.py:930]: There are other uses of `GbqConnector` that need to be updated. Please add these auth argument to `to_gbq` as well.

**Review by tswast — CHANGES_REQUESTED** [review]

File: pandas_gbq/gbq.py (hunk 4 above)
Comment (chalmerlowe) [reply pandas_gbq/gbq.py:930]: done

File: pandas_gbq/gbq.py (hunk 3 above)
Comment (chalmerlowe) [reply pandas_gbq/gbq.py]: Done. Also added the same docstrings to the `to_gbq()` function.

File: pandas_gbq/gbq.py [hunk 5]
```diff
@@ -864,6 +877,15 @@ def read_gbq(
         or
         :func:`google.oauth2.service_account.Credentials.from_service_account_file`
         instead.
+    auth_redirect_uri : str
+        Path to the authentication page for organization-specific authentication
+        workflows.
```
Comment (tswast) [inline pandas_gbq/gbq.py:882]:
```suggestion
        workflows. Used when `auth_local_webserver=False`.
```

File: pandas_gbq/gbq.py [hunk 6]
```diff
@@ -1071,6 +1099,15 @@ def to_gbq(
...
```
Comment (tswast) [inline pandas_gbq/gbq.py:1104]:
```suggestion
        workflows. Used when `auth_local_webserver=False`.
```

**Review by tswast — APPROVED** [review]

**Review by tswast — APPROVED** [review]

**Pull request merged**